tx-parser/src/lib.rs

```diff
@@ -30,181 +34,150 @@ fn fn_parser<O, I>(f: fn(I) -> ParseResult<O, I>, err: &'static str) -> TxParser
     parser(f).expected(err)
 }
 
-impl<I> Tx<I>
-    where I: Stream<Item = Value>
-{
-    fn integer() -> TxParser<i64, I> {
-        fn_parser(Tx::<I>::integer_, "integer")
-    }
+def_parser_fn!(Tx, integer, Value, i64, input, {
+    return satisfy_map(|x: Value| if let Value::Integer(y) = x {
+            Some(y)
+        } else {
+            None
+        })
+        .parse_stream(input);
+});
 
-    fn integer_(input: I) -> ParseResult<i64, I> {
-        return satisfy_map(|x: Value| if let Value::Integer(y) = x {
-                Some(y)
-            } else {
-                None
-            })
-            .parse_stream(input);
-    }
+// The above using a def_value_satisfy_parser_fn
+//def_value_satisfy_parser_fn!(Tx, integer, i64, if let Value::Integer(y) = x { Some(y) } else { None });
 
-    fn keyword() -> TxParser<NamespacedKeyword, I> {
-        fn_parser(Tx::<I>::keyword_, "keyword")
-    }
+// Manually expanding the above:
+// def_value_parser_fn!(Tx, integer, i64, input, {
+//     satisfy_map(|x: edn::Value| $transformer(&x)                                      ).parse_stream(input)
+//     satisfy_map(|x: edn::Value| if let Value::Integer(y) = x { Some(y) } else { None }).parse_stream(input)
```

Comment:
The trick is that `$transformer` is a path, not a block, so you can't pass an expression to this macro.